src/OrchardCore.Modules/OrchardCore.Email.Azure/Services/AzureEmailDeliveryService.cs

```diff
@@ -0,0 +1,140 @@
+        var client = new EmailClient(_emailSettings.ConnectionString);
+
+        try
+        {
+            var senderAddress = string.IsNullOrWhiteSpace(message.From)
+                ? _emailSettings.DefaultSender
+                : message.From;
+
+            if (!string.IsNullOrWhiteSpace(senderAddress))
+            {
+                message.From = senderAddress;
+            }
+
+            var emailMessage = FromMailMessage(message, out result);
+
+            await client.SendAsync(WaitUntil.Completed, emailMessage);
+
+            result = EmailResult.Success;
+        }
+        catch (Exception ex)
+        {
+            result = EmailResult.Failed(S["An error occurred while sending an email: '{0}'", ex.Message]);
+
+            _logger.LogError(ex, message: ex.Message);
+        }
+
+        return result;
+    }
+
+    private EmailMessage FromMailMessage(MailMessage message, out EmailResult result)
+    {
+        var recipients = message.GetRecipients();
+
+        List<EmailAddress> toRecipients = null;
+        if (recipients.To.Count > 0)
+        {
+            toRecipients = [.. recipients.To.Select(r => new EmailAddress(r))];
+        }
+
+        List<EmailAddress> ccRecipients = null;
+        if (recipients.Cc.Count > 0)
+        {
+            ccRecipients = [.. recipients.Cc.Select(r => new EmailAddress(r))];
+        }
+
+        List<EmailAddress> bccRecipients = null;
+        if (recipients.Bcc.Count > 0)
+        {
+            bccRecipients = [.. recipients.Bcc.Select(r => new EmailAddress(r))];
+        }
+
+        var content = new EmailContent(message.Subject);
+        if (message.IsHtmlBody)
+        {
+            content.Html = message.Body;
+        }
+        else
+        {
+            content.PlainText = message.Body;
+        }
+
+        var emailMessage = new EmailMessage(
+            message.From,
+            new EmailRecipients(toRecipients, ccRecipients, bccRecipients),
+            content);
+
+        foreach (var address in message.GetReplyTo())
+        {
+            emailMessage.ReplyTo.Add(new EmailAddress(address));
+        }
+
+        foreach (var attachment in message.Attachments)
+        {
+            // Stream must not be null, otherwise it would try to get the filesystem path
+            if (attachment.Stream != null)
+            {
+                var extension = Path.GetExtension(attachment.Filename);
+
+                if (_allowedMimeTypes.TryGetValue(extension, out var contentType))
+                {
+                    var data = new byte[attachment.Stream.Length];
+
+                    attachment.Stream.Read(data, 0, (int)attachment.Stream.Length);
+
+                    emailMessage.Attachments.Add(new EmailAttachment(attachment.Filename, contentType, new BinaryData(data)));
+                }
+                else
+                {
+                    result = EmailResult.Failed(S["Unable to attach the file named '{0}'.", attachment.Filename]);
```

Comment:
```suggestion
                    result = EmailResult.Failed(S["Unable to attach the file named '{0}' since its type is not supported.", attachment.Filename]);
```